src/main/java/md2k/mcerebrum/cstress/features/ECGDataQuality.java

```diff
@@ -70,4 +103,88 @@ public ECGDataQuality(DataStreams datastreams, double qualityThreshold) {
 
 
     }
+
+    private List<DataPoint> computeQuality(List<DataPoint> ecg, long windowSize) {
+
+        List<DataPoint[]> windowedECG = Time.window(ecg, windowSize);
+        List<DataPoint> result = new ArrayList<>();
+
+        for (DataPoint[] dpA : windowedECG) {
+            int[] data = new int[dpA.length];
+            int i = 0;
+            for (DataPoint s : dpA) {
+                data[i++] = (int) s.value;
+            }
+            if (data.length > 0) {
+                result.add(new DataPoint(dpA[0].timestamp, currentQuality(data)));
+            }
+        }
+
+
+        return result;
+
+    }
+    private int currentQuality(int[] data) {
+        classifyDataPoints(data);
+        classifySegment(data);
+
+        classBuff[(classHead++) % classBuff.length] = segment_class;
+        envelBuff[(envelHead++) % envelBuff.length] = max_value - min_value;
+        classifyBuffer();
```

Comment:
Methods should be written to not have side-effects.  This clearly does due to it not having a return value.